# This patch fixes a bunch of fails on Windows, which show up only with -O0.
# See ISPC bug 1454 for more details. 
Index: lib/Target/X86/X86ISelLowering.cpp
===================================================================
--- a/llvm/lib/Target/X86/X86ISelLowering.cpp	(revision 357656)
+++ b/llvm/lib/Target/X86/X86ISelLowering.cpp	(working copy)
@@ -3022,7 +3022,9 @@
   }
 
   // This is an argument in memory. We might be able to perform copy elision.
-  if (Flags.isCopyElisionCandidate()) {
+  // Unless its passed by pointer.
+  if (Flags.isCopyElisionCandidate() &&
+      VA.getLocInfo() != CCValAssign::Indirect && !ExtendedInMem) {
     EVT ArgVT = Ins[i].ArgVT;
     SDValue PartAddr;
     if (Ins[i].PartOffset == 0) {
